src/components/content/availableServices/tree/RegisteredServicesFullView.tsx

```diff
@@ -0,0 +1,163 @@
+import catalogStyles from '../../../../styles/catalog.module.css';
+import { ServiceTemplateDetailVo } from '../../../../xpanse-api/generated';
+import {
+    registeredServicesPageRoute,
+    serviceCategoryQuery,
+    serviceHostingTypeQuery,
+    serviceNameKeyQuery,
+    serviceNamespaceQuery,
+    serviceVersionKeyQuery,
+} from '../../../utils/constants.tsx';
+import { getFourthLevelKeysFromAvailableServicesTree } from '../../common/registeredServices/registeredServiceProps.ts';
+import { RegisteredServicesTree } from './RegisteredServicesTree.tsx';
+import ServiceContent from './ServiceContent.tsx';
+
+export function RegisteredServicesFullView({
+    treeData,
+    availableServiceList,
+}: {
+    treeData: DataNode[];
+    availableServiceList: ServiceTemplateDetailVo[];
+}): React.JSX.Element {
+    const [urlParams] = useSearchParams();
+
+    const serviceNamespaceInQuery = useMemo(() => {
+        const queryInUri = decodeURI(urlParams.get(serviceNamespaceQuery) ?? '');
+        if (queryInUri.length > 0) {
+            return queryInUri;
+        }
+        return '';
+    }, [urlParams]);
+
+    const serviceCategoryInQuery = useMemo(() => {
+        const queryInUri = decodeURI(urlParams.get(serviceCategoryQuery) ?? '');
+        if (queryInUri.length > 0) {
+            return queryInUri;
+        }
+        return '';
+    }, [urlParams]);
+
+    const serviceNameInQuery = useMemo(() => {
+        const queryInUri = decodeURI(urlParams.get(serviceNameKeyQuery) ?? '');
+        if (queryInUri.length > 0) {
+            return queryInUri;
+        }
+        return '';
+    }, [urlParams]);
+
+    const serviceVersionInQuery = useMemo(() => {
+        const queryInUri = decodeURI(urlParams.get(serviceVersionKeyQuery) ?? '');
+        if (queryInUri.length > 0) {
+            return queryInUri;
+        }
+        return '';
+    }, [urlParams]);
+
+    const navigate = useNavigate();
+    const allKeysInTree: React.Key[] = getFourthLevelKeysFromAvailableServicesTree(treeData);
+
+    const [selectKey, setSelectKey] = useState<React.Key>(getDefaultSelectedKey());
+    // necessary to store previous selected key to check and reuse previous values where possible.
+    const previousSelectedKey = useRef<React.Key>(getDefaultSelectedKey());
+
+    function getDefaultSelectedKey() {
+        //if user directly opens an url.
+        if (serviceNamespaceInQuery && serviceCategoryInQuery && serviceNameInQuery && serviceVersionInQuery) {
+            const fullKey = `${serviceNamespaceInQuery}@${serviceCategoryInQuery}@${serviceNameInQuery}@${serviceVersionInQuery}`;
+
+            if (allKeysInTree.includes(fullKey)) {
+                return fullKey;
+            }
+        }
+        return allKeysInTree[0];
+    }
+
+    // useEffect necessary since we are updating URL outside the React context.
+    useEffect(() => {
+        if (allKeysInTree.length > 0) {
```

Comment:
do we really need this check again? if selectKey is updated means, the allKeysTree is not empty know?